ext/com_dotnet/com_extension.c

```diff
@@ -187,84 +187,6 @@ PHP_MINIT_FUNCTION(com_dotnet)
 
 	REGISTER_INI_ENTRIES();
 
-#define COM_CONST(x) REGISTER_LONG_CONSTANT(#x, x, CONST_CS|CONST_PERSISTENT)
-
-#if SIZEOF_ZEND_LONG == 8
-# define COM_ERR_CONST(x) REGISTER_LONG_CONSTANT(#x, (zend_long) (ULONG) (x), CONST_CS|CONST_PERSISTENT)
```

Comment:
The values are of type `HRESULT` which is defined as `LONG` (i.e. a signed 32bit value on both x64 and x86). However, the values are usually expressed as unsigned (e.g. `DISP_E_DIVBYZERO` is `0x80020012`), so we need to cast on x64 to avoid having a negative number there. To not break BC, we need to keep the casts (the `(zend_long)` cast may actually not be needed, but might suppress a compiler warning).

Reply:
Thank you for the answer! I've just pushed another version which should fix these issues. What do you think? Is there a more elegant solution?

Reply:
LGTM as is.